lms/djangoapps/onboarding_survey/views.py

```diff
@@ -242,3 +268,47 @@ def get_languages(request):
     mime_type = 'application/json'
 
     return HttpResponse(data, mime_type)
+
+
+def update_account_settings(request):
+    """
+    View to handle update of registration extra fields
+    """
+    if request.method == 'POST':
+        user_extended_profile = request.user.extended_profile
+        form = forms.UpdateRegModelForm(request.POST, instance=user_extended_profile)
+        if form.is_valid():
+            form_instance = form.save(commit=False)
+            if form_instance.is_poc:
+                form_instance.org_admin_email = ""
+            form_instance.save()
+
+    else:
+        user_extended_profile = request.user.extended_profile
+        form = forms.UpdateRegModelForm(
+            instance=user_extended_profile,
+            initial={
+                'organization_name': user_extended_profile.organization.name,
+                'is_poc': 1 if user_extended_profile.is_poc else 0
+            }
+        )
+
+    return render(
+        request, 'onboarding_survey/registration_update.html',
+        {'form': form, 'org_url': reverse('get_user_organizations')}
+    )
+
+
+@csrf_exempt
+def get_user_organizations(request):
+    """
+    Get organizations
+    """
+    final_result = {}
+    if request.is_ajax():
+        query = request.GET.get('term', '')
+        all_organizations = Organization.objects.filter(name__startswith=query)
+        for organization in all_organizations:
```

Comment:
what is the reason for iterating the queryset here?

Reply:
We need to iterate over the queryset so that we can create a custom json for the view. The view will show certain fields whether admin for the org exists or not.